drivers/storage/portworx/preflight.go

```diff
@@ -208,8 +209,12 @@ func (u *preFlightPortworx) RunPreFlight() error {
 		logrus.Infof("runPreflight: running pre-flight with existing PX-StoreV2 param, hard fail check enabled")
 	}
 
+	preFltEndPtPort := component.GetCCMListeningPort(u.cluster) + 1 // preflight endpoint port  +1 CCM uploader port
+	logrus.Infof("runPreflight: Setting port: %d", preFltEndPtPort)
+	preflightDS.Spec.Template.Spec.Containers[0].ReadinessProbe.ProbeHandler.HTTPGet.Port = intstr.FromInt(preFltEndPtPort)
```

Comment:
wow.. you're putting a lot of faith in this code -- if any of the 8 `.<object>` happens to be `nil` -- this will cause a panic.

Are we guaranteed that all sub-objects can't be `nill` (and also `len(.Containers) > 1`)?

Reply:
This is the OCI-Mon container spec created by the deployment.go code ensures those exists.   I can add a routine with checks but the main thing I want to do is change the port.  In fact there will be a change coming that just generate pre-flight spec using the same routines that generate the OCI-Mon spec.   So we will be will basically only starting one container.. cause right now we also start the csi container along with this.   I will add some checks in there to be safe 👍.

Reply:
Oh and the initial objects "preflightDS.Spec.Template.Spec" are actually created above.   So I will check the passed in podSpec.. Thanks Z...